examples/asr/speech_to_text.py

```diff
@@ -51,7 +51,31 @@ def main(args):
     model_config['AudioToTextDataLayer_eval']['manifest_filepath'] = args.eval_dataset
     asr_model.setup_training_data(model_config['AudioToTextDataLayer'])
     asr_model.setup_validation_data(model_config['AudioToTextDataLayer_eval'])
-    asr_model.setup_optimization(optim_params={'optimizer': args.optimizer, 'lr': args.lr, 'opt_args': args.opt_args})
+
+    # Setup optimizer and scheduler
+    scheduler_args = {
+        'monitor': 'val_loss',  # pytorch lightning requires this value
+        'warmup_ratio': args.warmup_ratio,
+        'warmup_steps': args.warmup_steps,
+        'min_lr': args.min_lr,
+        'last_epoch': args.last_epoch,
+    }
+
+    if args.max_steps is None:
+        iters_per_sample = args.max_epochs / float(args.gpus * args.num_nodes * args.accumulate_grad_batches)
```

Comment:
shouldnt batch size be also in the denominator? 
this term is so weird, since it only makes sense when multiplied with len(dataset) which is done later. but ok

Reply:
Ok so batch size is not here, but deferred to the train_dataloader. Renamed it to `iters_per_batch` instead.